apis/Google.Cloud.Firestore/Google.Cloud.Firestore/ValueSerializer.cs

```diff
@@ -119,8 +119,10 @@ internal static Value Serialize(object value)
             GaxPreconditions.CheckNotNull(value, nameof(value));
             switch (value)
             {
+                case IDictionary<string, Value> map:
+                    return map.ToDictionary(pair => pair.Key, pair => pair.Value);
```

Comment:
Why not just return `map` here so you don't need to create a new collection?

Reply:
@evildour It was a comment from my previous request

https://github.com/GoogleCloudPlatform/google-cloud-dotnet/pull/1795#issuecomment-354103718

Reply:
I see. Thanks.